func.go

```diff
@@ -163,6 +164,16 @@ func RegisterFunc(fptr interface{}, cfn uintptr) {
 				panic("purego: unsupported kind " + arg.Kind().String())
 			}
 		}
+		if ty.NumOut() > 0 && ty.Out(0).Kind() == reflect.Struct {
+			if runtime.GOOS != "darwin" {
+				panic("purego: struct return values only supported on darwin arm64 & amd64")
+			}
+			outType := ty.Out(0)
+			checkStructFieldsSupported(outType)
+			if outType.Size() > 16 {
+				ints++
+			}
```

Comment:
It seems that for a structure like `struct{a, b bool}` we won't increment amount of occupied integer registers, is this intentional?

Reply:
Oh, I get it now. We use additional parameter to pass a pointer for a struct more than 16 bytes in size. It's nitpicking, but I'd really love a comment about, why we use an int register to receive a return value

Reply:
Having a comment to explaing why 16 sounds nice.

Reply:
Also, 16 is a magic number. Let's define a constant and use it everywhere.